### DISCUSSION TITLE
Servo Plugin - for Pen Plotter

### DISCUSSION THREAD

**Discussion by ghost** [grblHAL/plugins#7] [Ideas]
Hi Team,

I was wondering if anyone has thought about creating a Servo Plugin that can be used for a Pen Plotter.

I can build the hardware but do not know how to program the code to support a servo output.

Regards
George

**Comment by terjeio** [comment]
Check the [drivers readme](https://github.com/grblHAL/drivers) for _RC Servo/ESC for spindle_ support:

> <sup>13</sup> Set $33=50 (PWM frequency), $34=5, $35=5 and $36=10 to generate a "standard" PWM signal: 20ms repetition rate, 1 - 2ms pulse length range.

**Comment by ghost** [comment]
Hi terjeio,

Thanks for pointing this out. 

I found that it did not work until I changed the frequency to 61Hz. Playing around with the min / max levels allowed for the servor arm to travel further.

Cheers
George

**Reply by terjeio** [reply]
Which driver are you using?

**Comment by ghost** [comment]
STM32F401.

After tweeking the settings the servo works really well.

Cheers
George
